src/kubernetes/utils.ts

```diff
@@ -39,3 +44,41 @@ export const extractNodeLabels = (
     [label.key]: label.value,
   };
 };
+
+/**
+ * Common labels used in the daily.dev infrastructure.
+ * These labels are used to identify resources created by pulumi, and of which version of the pulumi-common library was used to create them.
+ */
+export const commonLabels = {
+  'app.kubernetes.io/managed-by': 'pulumi',
+  'pulumi-common.daily.dev/version': version,
+} as const;
+
+/**
+ * Returns the image string to be used in a Kubernetes deployment.
+ * If a digest is provided, it will be used instead of the tag.
+ */
+export const image = ({ repository, tag, digest }: Image) => {
+  const separator = digest ? '@' : ':';
+  return `${repository}${separator}${digest || tag}`;
+};
+
+export const configureResources = (
+  args: AdhocEnv & {
+    resources?: Resources;
+  },
+): Output<Resources> | undefined => {
+  return args.isAdhocEnv
+    ? undefined
+    : all([args.resources]).apply(([resources]) => {
+        return {
+          requests: {
+            cpu: resources?.requests?.cpu || '1000m',
+            memory: resources?.requests?.memory || '1Gi',
+          },
+          limits: {
+            memory: resources?.limits?.memory || '2Gi',
+          },
```

Comment:
These defaults look pretty high. Should we have defaults even?

Reply:
yap we can even throw an exception (but we should make sure it doesn't break existing projects somehow)

Reply:
It wouldn't break existing projects, as this is a new function, and only used in the clickhouse sync deployment at the moment.